runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/Exceptions.kt

```diff
@@ -88,6 +89,12 @@ open class ServiceErrorMetadata : ErrorMetadata() {
     val serviceName: String
         get() = attributes.getOrNull(SdkClientOption.ServiceName) ?: ""
 
+    /**
+     * Returns the error code associated with the response
+     */
+    val errorCode: String?
```

Comment:
suggestion

We should document that this and `requestId` are implementation defined.

e.g.

```
Returns the error code associated with the response (if available).

This value is implementation defined, for example AWS protocols use error code to identify the shape ID of the error (see [restJson1 protocol errors](https://awslabs.github.io/smithy/1.0/spec/aws/aws-restjson1-protocol.html#operation-error-serialization))
```